stacks-signer/src/signer.rs

```diff
@@ -1328,6 +1360,74 @@ impl Signer {
         Ok(())
     }
 
+    /// Load the saved signer state for the current dkg round
+    pub fn load_saved_state(&mut self) -> Result<(), PersistenceError> {
+        let Some(aggregate_key) = self.approved_aggregate_public_key else {
+            return Ok(());
+        };
+
+        info!("{self}: Loading saved state for key: {aggregate_key}");
+        if let Some(state) = self.load_saved_state_for_aggregate_key(aggregate_key)? {
+            let party_id = state.party_id;
+            let poly_commitment = state.get_poly_commitment(&mut OsRng);
+
+            let party_polynomials = poly_commitment
+                .as_ref()
+                .map(|poly_commitment| (&party_id, poly_commitment));
+
+            let mut dkg_results_bytes = vec![];
+
+            if let Err(e) = SignerMessage::serialize_dkg_result(
+                &mut dkg_results_bytes,
+                &aggregate_key,
+                party_polynomials.into_iter(),
+            ) {
+                error!(
+                    "{self}: Failed to serialize DKGResults message after loading saved state: {e}"
+                );
+            } else if let Err(e) = self
+                .stackerdb
+                .send_message_bytes_with_retry(&MessageSlotID::DkgResults, dkg_results_bytes)
+            {
+                error!("{self}: Failed to send DKGResults message to StackerDB after loading: {e}");
+            };
+
+            self.state_machine.signer = state;
+        } else {
+            warn!("{self}: Signer unable to load state for key {aggregate_key}");
+        };
+
+        Ok(())
+    }
+
+    /// Load the saved state for a particular aggregate key.
+    fn load_saved_state_for_aggregate_key(
+        &mut self,
+        aggregate_key: Point,
+    ) -> Result<Option<v2::Signer>, PersistenceError> {
+        Ok(
+            get_signer_state(self.load_encrypted_signer_states()?, aggregate_key)
+                .map(|state| v2::Signer::load(&state)),
+        )
+    }
+
+    /// Load the entire encrypted signer state
+    fn load_encrypted_signer_states(&mut self) -> Result<StoredSignerStates, PersistenceError> {
+        let loaded_signers = load_encrypted_signer_state(
+            &mut self.stackerdb,
+            self.signer_slot_id.into(),
+            &self.state_machine.network_private_key,
+        ).or_else(|err| {
+                warn!("Failed to load encrypted signer state from StackerDB, falling back to SignerDB: {err}");
+                load_encrypted_signer_state(
+                    &self.signer_db,
+                    self.reward_cycle,
+                    &self.state_machine.network_private_key)
+        })?;
```

Comment:
I'm curious about the ordering here. Loading the state from signer_db doesn't require RPC calls while loading from stackerdb does. Even in a perfect world, RPCs require a inter-process call at least, but the stacks-node RPC interface isn't perfect and could definitely stall while doing things like processing blocks, etc. It seems preferable to query the signer_db first -- is there a good reason not to?

Reply:
Changed the ordering.